### PR TITLE
fix: change `Component.Evidence.Identity` to array

### LINKED ISSUE #192: Improve support for the `Component.Evidence.Identity` field in CycloneDX 1.6
In CycloneDX 1.5, the `Component.Evidence.Identity` field was specified as a single `Identity` object. In CycloneDX 1.6 this has been deprecated in favor of an array of `Identity` objects.
The specifications can be compared here:
1.5: https://cyclonedx.org/docs/1.5/json/#components_items_evidence_identity
1.6: https://cyclonedx.org/docs/1.6/json/#components_items_evidence_identity

cyclonedx-go still defines `Identity` as `*EvidenceIdentity` so it fails to unmarshal SBOMs which have an array of identities in place of a single one; the error is: `cannot unmarshal array into Go struct field Evidence.components.evidence.identity of type cyclonedx.EvidenceIdentity`.

This currently happens with some SBOMs generated by cdxgen (https://github.com/CycloneDX/cdxgen) when using the `--spec-version 1.6` argument. [This is an SBOM](https://github.com/user-attachments/files/16426343/bom_1.6.json) affected by the issue. It was generated from a skeleton Poetry project, with just pytest added, using this command:
```
docker run --rm -v /tmp:/tmp -v $(pwd):/app:rw -t ghcr.io/cyclonedx/cdxgen -r /app -o /app/bom.json --spec-version 1.6`
```

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
in CycloneDX 1.6 type of `Component.Evidence.Identity` has been changed to `[]Identity`.
`Identity` as `struct` was marked as `deprecated`.
https://cyclonedx.org/docs/1.6/json/#tab-pane_metadata_component_evidence_identity_oneOf_i0

## Related issues
- Close #192

### CODE REVIEW

File: convert.go [hunk 1]
```diff
@@ -195,6 +195,13 @@ func convertEvidence(c *Component, specVersion SpecVersion) {
 	}
 
 	if specVersion < SpecVersion1_6 {
+		// Spec version 1.5 uses only one Identity.
+		// cf. https://cyclonedx.org/docs/1.5/json/#components_items_evidence_identity
+		if c.Evidence.Identity != nil {
+			ids := *c.Evidence.Identity
+			ids = ids[:1]
+			c.Evidence.Identity = &ids
+		}
```
Comment (DmitriyLewen) [inline convert.go:204]: I'm not sure about this way...
Maybe we need to select one `Identifier` using `Field` with some order

Please tell me if you have a better solution.

File: cyclonedx_xml_test.go
Comment (DmitriyLewen) [inline cyclonedx_xml_test.go:1]: I overwrite all the Evidence fields in the `MarshalXML` and `UnmarshalXML` functions, so I added test cases for each field.

Tell me if this is too much

**Review by nscuro — APPROVED** [review]

File: convert.go (hunk 1 above)
Comment (nscuro) [reply convert.go:204]: IMO that is fine for now, since it keeps the insertion order.

**Comment by nscuro** [comment]
Hmmm, seems like CI never ran for his PR and I can't manually trigger it. @DmitriyLewen do you mind re-opening this PR, or alternatively amend your last commit so GitHub triggers a build?

**Comment by DmitriyLewen** [comment]
Hi @nscuro
I reran actions (I merged `master` branch), but this action still doesn't run despite having your approval - https://github.com/CycloneDX/cyclonedx-go/actions/runs/12270387292
Do you have ideas?

**Comment by nscuro** [comment]
@DmitriyLewen I had to approve CI to run since this is your first PR for this repo. However, I wasn't given that option before you pushed a new commit, which is odd.

Now we have a few test failures.

**Comment by DmitriyLewen** [comment]
Hello @nscuro 
Thanks!

I fixed [TestEvidence_MarshalJSON/WithIdentify](https://github.com/CycloneDX/cyclonedx-go/actions/runs/12270387292/job/34251289295#step:4:84) in https://github.com/CycloneDX/cyclonedx-go/pull/204/commits/24e9503293f0837e6e7ea3ff670ef958e6075b87

About [TestRoundTripJSON/valid-evidence.json](https://github.com/CycloneDX/cyclonedx-go/actions/runs/12270387292/job/34251289295#step:4:363):
I forgot to update `valid-evidence.json` and updated this file in https://github.com/CycloneDX/cyclonedx-go/pull/204/commits/d68a199bc1747e5d6a7d4196c2f270535bbf6e3e.
But this test is still fail (because schema for `identity` has been changed). 
What can i do with this test?

**Comment by DmitriyLewen** [comment]
Got it. Thanks for your help!

All tests passed:
```
PASS
coverage: 83.0% of statements
ok      github.com/CycloneDX/cyclonedx-go       37.215s

```

**Review by nscuro — APPROVED** [review]
Thanks!

**Pull request merged**